webapp/components/at_mention/at_mention.jsx

```diff
@@ -48,32 +72,54 @@ export default class AtMention extends React.PureComponent {
         return '';
     }
 
-    search = (e) => {
-        e.preventDefault();
+    getProfileByUsername(username) {
```

Comment:
Maybe combine these two functions by changing `getUsernameFromMentionName` to return the user object instead of just the username? If `getUsernameFromMentionName` returns anything, it's because it's found a user already, so we don't need to go to the `UserStore` or call `getUserByUsername`